x-pack/plugins/cases/public/components/actions/tags/use_tags_action.tsx

```diff
@@ -33,20 +34,36 @@ export const useTagsAction = ({ onAction, onActionSuccess, isDisabled }: UseActi
     [onAction]
   );
 
+  const areTagsEqual = (originalTags: string[], tagsToUpdate: Set<string>): boolean => {
+    if (originalTags.length !== tagsToUpdate.size) {
+      return false;
+    }
+
+    return originalTags.every((tag) => tagsToUpdate.has(tag));
+  };
+
   const onSaveTags = useCallback(
     (tagsSelection: TagsSelectionState) => {
       onAction();
       onFlyoutClosed();
-      const casesToUpdate = selectedCasesToEditTags.map((theCase) => {
-        const tags = difference(theCase.tags, tagsSelection.unSelectedTags);
-        const uniqueTags = new Set([...tags, ...tagsSelection.selectedTags]);
 
-        return {
-          tags: Array.from(uniqueTags.values()),
-          id: theCase.id,
-          version: theCase.version,
-        };
-      });
+      const casesToUpdate = selectedCasesToEditTags.reduce((acc, theCase) => {
+        const tagsWithoutUnselectedTags = difference(theCase.tags, tagsSelection.unSelectedTags);
```

Comment:
If you make this one a set too you can just use lodash's `.isEqual()`

Reply:
Done!